test/test_before_build.py

```diff
@@ -24,6 +24,12 @@
         print('sys.executable', sys.executable)
         # windows/mac are case insensitive
         assert os.path.realpath(stored_executable).lower() == os.path.realpath(sys.executable).lower()
+
+        if sys.platform == 'linux':
+            cwd_file = '/tmp/cwd.txt'
+            with open(cwd_file) as f:
+                stored_cwd = f.read()
+            assert stored_cwd == '/project'
```

Comment:
Looks like this is missing a mac and windows test. Also, it might be a bit Rube Goldberg - could you instead `assert os.getcwd() == '/project'`? And in BEFORE_ALL, do the same?

Reply:
> Looks like this is missing a mac and windows test.

Well, yes, but I didn't know what directory they're supposed to be executed from? As far as I know, it will depend on the CI; probably somewhere in the home directory or a subdirectory?

> Also, it might be a bit Rube Goldberg - could you instead `assert os.getcwd() == '/project'`?

Nope, that's not working. The tests are actually executed from a different directory, and we want to test `CIBW_BEFORE_BUILD` is executed from the right directory. Or do you mean to `python -c "import os;  os.getcwd() == '/project'"` in `CIBW_BEFORE_BUILD`?

> And in BEFORE_ALL, do the same?

Yes, will do. But I'll wait until the above gets settled.

Reply:
> Or do you mean to `python -c "import os; os.getcwd() == '/project'"` in `CIBW_BEFORE_BUILD`?

Yeah, that's what I was thinking, I suppose it'd be something like...
```
{
   'CIBW_BEFORE_BUILD': f'''python -c "import os; assert os.getcwd() == {os.getcwd()!r}"''',
   'CIBW_BEFORE_ALL': f'''python -c "import os; assert os.getcwd() == {os.getcwd()!r}"''',
   'CIBW_BEFORE_BUILD_LINUX': '''python -c "import os; assert os.getcwd() == '/project'"''',
   'CIBW_BEFORE_ALL_LINUX': '''python -c "import os; assert os.getcwd() == '/project'"''',
}
```

Reply:
Windows on GH Actions uses `/d/a/bls-signatures/bls-signatures` which is the repo name twice: https://github.com/Chia-Network/bls-signatures/runs/887873992?check_suite_focus=true#step:10:60

Mac is `/Users/runner/work/bls-signatures/bls-signatures/` see: https://github.com/Chia-Network/bls-signatures/runs/887873983?check_suite_focus=true#step:11:535

Reply:
> Windows on GH Actions uses `/d/a/bls-signatures/bls-signatures` which is the repo name twice: https://github.com/Chia-Network/bls-signatures/runs/887873992?check_suite_focus=true#step:10:60
> 
> Mac is `/Users/runner/work/bls-signatures/bls-signatures/` see: https://github.com/Chia-Network/bls-signatures/runs/887873983?check_suite_focus=true#step:11:535

Thanks, but that's different from every CI and every project, even (since it contains the project name), so that doesn't seem feasible to distinguish. Let me try @joerick's plan (maybe with `__file__` rather than `getcwd`?) :-)

Reply:
OK, like this?

I added a test that checks whether a failing `CIBW_BEFORE_BUILD` and `CIBW_BEFORE_ALL` also fail the whole run.

And I (temporarily) reverted the actual fix, to see if these tests catch the errors. Can be dropped in a rebase, later.